src/server.ts

```diff
@@ -24,7 +24,7 @@ const app: Application = express();
       baseDir: __dirname, // defaults to `process.cwd()`
   })
   DI.em = DI.orm.em;
-  DI.planetRepository = DI.orm.em.getRepository(Planet);
+  DI.planetRepository = DI.orm.em.getRepository('Planet') as unknown as EntityRepository<Planet>;
```

Comment:
This is weird strict typing....

Reply:
Digging a little bit deeper, this might be a cleaner approach:

```suggestion
  DI.planetRepository = DI.orm.em.getRepository<Planet>('Planet');
```

The problems here started with this:

```typescript
  DI.planetRepository = DI.orm.em.getRepository(Planet);
  ^^^^^^^^^^^^^^^^^^^                           ^^^^^^
  1                                             2
```
Two `typescript` errors crop up (underlined, labelled above). I'll unpack both of them below:

1. `Type 'EntityRepository<AnyEntity<unknown>>' is not assignable to type 'EntityRepository<Planet>'.`

   So, here we have a classic generic programming problem. The type defined by `EntityManager::getRepository` looks like this:
   ```typescript
   getRepository<
     T extends AnyEntity<T>,
     U extends EntityRepository<T> = EntityRepository<T>
   >(entityName: EntityName<T>): GetRepository<T, U>;
   ```

   All that really means is that without you explicitly providing generic types when you call the method (like this, `getRepository<SomeType, SomeOtherType>(...)`), `typescript` is going to try to infer which type is being returned based on the input argument to the method. In this case, switching the argument to a `string` allows us to ignore the error of providing an `interface` (explained below) but this leads to a problem where the compiler now can't infer the type, because `mikro-orm` is going to fetch the repository based on the name `string` that you provided and that could be anything at runtime (`typescript` labels this as `unknown`). To solve this, we just need to tell the compiler, "trust me, at runtime this will be a repository for type `Planet`". From there, casting or providing a type parameter before the input arguments do the same thing: tell the compiler the type involved.

2. `'Planet' only refers to a type, but is being used as a value here.ts(2693)`

   As we solved this above, `mikro-orm` is nice enough to allow the use of `interface`s as entities, however, in `typescript` `interface` is erased at runtime (much like generic types are in Java). Therefore, it is of no use to pass it as a value to `getRepository`, since it will be removed after compilation, leaving this method without any arguments provided.

   `mikro-orm` also has a solution there: just pass the `string` name of the entity. In doing so, we lose type-safety (we never had it anyway since passing an `interface` is invalid) and that leads to the error from above.